docs/proposals/20231122-security-groups.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Flexible API for Managed Security Groups
+authors:
+  - "@emilienm"
+reviewers:
+  - "@jichenjc"
+  - "@mdbooth"
+creation-date: 2023-11-22
+last-updated: 2023-11-22
+
+status: provisional
+---
+
+## Title
+
+Flexible API for Managed Security Groups
+
+## Table of Contents
+
+- [New API for managing Security Groups](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Security Group - A security group is a named set of rules that control traffic to and from virtual machine instances. The rules can specify which ports and protocols to allow, and which IP address ranges to allow traffic to or from. Security groups can be associated with multiple instances, and multiple security groups can be associated with a single instance. Security groups act as a firewall for associated instances, controlling both inbound and outbound traffic at the instance level.
+
+## Summary
+
+This proposal introduces a new API to manage OpenStack Security Groups in a flexible way that will replace legacy rules that are can optionaly be managed by CAPO.
+
+## Motivation
+
+We want users to be able to manage their security groups in a flexible way for both the Control Plane and the Worker Nodes without enforcing unnecessary rules that are not needed by the user.
+
+### Goals
+
+1. Make the feature configurable via a new API
+2. Avoid causing and breaking changes to previous users, they'll still be able to use the legacy rules
```

Comment:
will this still be true based on our previous discussion?

Reply:
After some discussion with Matt, i've reworked the whole KEP. Please have a look again.